tests/neg/annot-printing.scala

```diff
@@ -0,0 +1,6 @@
+import scala.annotation.*
+class Foo() extends Annotation
+class Bar(s: String) extends Annotation
+
+def x: Int @nowarn @main @Foo @Bar("hello") = "abc" // error
```

Comment:
with these changes `-Xprint:parser` will only show `<none>` where the class name should be
```scala
def x: Int @<none> @<none> @<none> @<none>("hello") = "abc"
```

Reply:
in 3.1.0-RC1 we still show the correct class names

Reply:
Good catch! Is there an easy way to write a printing test to guard against regressions for this?

Reply:
the `PrintingTest` can be modified to show at parser instead:
```
compiler/test/dotty/tools/dotc/printing/PrintingTest.scala
```

Reply:
@odersky I have added tests to `PrintingTest` in latest commit